### ISSUE TITLE
Empfangszähler für Kontaktids auf der Oberfläche der Anwendung

### ISSUE THREAD

**Issue by SchwertAs** [corona-warn-app/cwa-wishlist#280]
## Feature description
Display dayly and weekly Contact-Amount of user on base of received IDs (infected or not).

Pseudocode:
Während eines Abfragescans:
   Wenn neuer Tag:
      setzte Tageszähler auf 0
   Wenn neue Woche:
      übertrage Wochenzähler auf Vorwochenzähler
      setze Wochenzähler auf 0
   Wenn Id empfangen (neu oder bekannt ist egal)
      erhöhe aktuellen Wochenzähler
      erhöhe aktuellen Tageszähler
  Zeige neuen Tageszähler an
  Zeige neuen Wochenzähler an
  Zeige Vorwochenzähler an

**Comment by Ein-Tim** [comment]
Just a small remark:
The number of non positive contacts is not revealed by the ENF (the underlying API from Apple/Google).
But since this is the Wishlist.

---

Related: #205

**Comment by dl1chb** [comment]
For me this counter could also act as a technical feedback: I do not know if everyone here has the app when going to the supermarked. I also cannot proof everything here is somehow working. So for me it would have the advantage to see, okay, it sees others using the app or it does not. Or even nobody else is using the app. Then I can see how much I can trust this "low risk" statement. Or maybe there is another way to do this (would be another feature request then). I honestly don't believe I never came in contact with no infected person during these days...

I am against having this as a proof of good behaviour. Sometimes I need food and need to go to the supermarked. Also there are people who cannot avoid contacts during their work (think for example of those who are working in supermarkets). Therefor it should need some passphrase to see this number.

**Comment by jucktnich** [comment]
If it would go live (I don't think so, because of the ENF) the counter should NOT be live, it should get updated every day, because there will be cries out of the media, because you see whether or not a person use the app and they'll think the voluntary of the app is affected.

**Comment by heinezen** [comment]
Related: #262

**Comment by heinezen** [comment]
Hello @SchwertAs ,

Thank you for the request. We have submitted your proposal to our internal Jira system (ticket ID: EXPOSUREAPP-4056) and forwarded it to the development team. The developers and the RKI will now decide if this feature can be implemented and how it would work with the other related requests. We will notify everyone on Github when we have something new to report about the status of the request.

Best Regards,
CH

----
Corona-Warn-App Open Source Team

**Comment by SchwertAs** [comment]
You did misunderstand my suggestion. I wanted to sum up ALL Contacts not only the positive ones.

 

Von: Ein_Tim <notifications@github.com> 
Gesendet: Freitag, 27. November 2020 07:57
An: corona-warn-app/cwa-wishlist <cwa-wishlist@noreply.github.com>
Cc: SchwertAs <andreas.macher@schwertas.de>; Author <author@noreply.github.com>
Betreff: Re: [corona-warn-app/cwa-wishlist] Empfangszähler für Kontaktids auf der Oberfläche der Anwendung (#280)

 

Just a small remark:
The number of non positive contacts is not revealed by the ENF (the underlying API from Apple/Google).
But since this is the Wishlist.

  _____  

Related: #205 <https://github.com/corona-warn-app/cwa-wishlist/issues/205>

**Comment by SchwertAs** [comment]
It would be sufficient to delay the counter display by a randomly time of some minutes to ensure this.

 

Von: jucktnich <notifications@github.com> 
Gesendet: Freitag, 27. November 2020 17:11
An: corona-warn-app/cwa-wishlist <cwa-wishlist@noreply.github.com>
Cc: SchwertAs <andreas.macher@schwertas.de>; Author <author@noreply.github.com>
Betreff: Re: [corona-warn-app/cwa-wishlist] Empfangszähler für Kontaktids auf der Oberfläche der Anwendung (#280)

 

If it would go live (I don't think so, because of the ENF) the counter should NOT be live, it should get updated every day, because there will be cries out of the media, because you see whether or not a person use the app and they'll think the voluntary of the app is affected.

**Comment by Ein-Tim** [comment]
Yeah IK @SchwertAs, but inside the ENF this is not possible because the Corona-Warn App only knows about the positive encounters, how many (or if you had) non-positive encounters is only known by the ENF, which does not *tell* this number to CWA.

**Comment by SchwertAs** [comment]
I f ENF does not tell maybe there is a way to ask it?

 

Von: Ein_Tim <notifications@github.com> 
Gesendet: Montag, 30. November 2020 08:38
An: corona-warn-app/cwa-wishlist <cwa-wishlist@noreply.github.com>
Cc: SchwertAs <andreas.macher@schwertas.de>; Mention <mention@noreply.github.com>
Betreff: Re: [corona-warn-app/cwa-wishlist] Empfangszähler für Kontaktids auf der Oberfläche der Anwendung (#280)

 

Yeah IK @SchwertAs <https://github.com/SchwertAs> , but inside the ENF this is not possible because the Corona-Warn App only knows about the positive encounters, how many (or if you had) non-positive encounters is only known by the ENF, which does not tell this number to CWA.

**Comment by Ein-Tim** [comment]
> I f ENF does not tell maybe there is a way to ask it?

No, it's not that easy. The ENF (Exposure Notification Framework) is developed by Apple & Google and also maintained by them. The German Corona-Warn App is _talking_ to the API to get the information you see in the App. 
There is some information which is _told_ to CWA by ENF, but is not shown to the user.
How many contacts (positive+negative encounters) the phone recorded  is not even _told_ to CWA by ENF.
So for your proposal (which IMHO is a duplicate of #5) Apple/Google would have to update the ENF.

But since this is the Wishlist, this can stay open, I just wanted to clarify that this is at the moment not possible to introduce.

**Comment by Ein-Tim** [comment]
Version 1.12 will contain some statistics (https://github.com/corona-warn-app/cwa-app-ios/pull/1807). 🎉

**Comment by di-sc** [comment]
> Mit dieser Zählweise bildet man eine generelle RIsikozahl, die sowohl Kontaktdauer, als auch Anzahl der Kontakte beinhaltet. Diese kann für den User ein wertvolles Feedback sein, wie hoch sein persönliches Risiko aufgrund seines eigenen Verhaltens bereits gestiegen ist. Durch Vorweisen einer entsprechend niedrigen Contact-Amount Zahl gegenüber Anderen ließe sich vorbildliches Verhalten und damit Vertrauen in verantwortliches Verhalten aufbauen.

If at some time Apple and/or Google will extend the ENF to provide the necessary data I would suggest to think about a real risk calculation. Because of the infection rate everyone on his/her own has a risk to be infected when meeting a person, independent of an existing positive test (sounds strange but unfortunately it's a fact). Using the current 7 day infection rate (best would be the local infection rate of the county) provided by the RKI, the infection rate seven days ago, the current R value, and the R value 7 days ago and the number of  higher risk contacts of the CWA a probability value can be calculated to be infected (has to be proved and optimized by experts, epidemiologists and statisticians, for sure).

E.g.: ` (IR7 * R7 + IR7_7 * R7_7) * number_of_high_risk_contacts_over_last_14_days`

This probability value can then be shown as a slider on the main page of the CWA or, what I would prefer shown as a **yellow notification**, indicating a high _activity risk_, when it is above a to be defined threshhold.
The advantage of this method is the dependence on the infection rate. The allowed Activity can be higher if the infection rate is low and Activity has to be reduced if the infection rate is high. This is what we want to be able to control the pandemic!
The "**yellow notification**" should then be accompanied with the hint to reduce contacts, not necessarily to make a test.

(@Vineet-Bhola: I would then change your notification color of issue [#431 2nd Level Contact tracing](https://github.com/corona-warn-app/cwa-wishlist/issues/431) (#24) to orange.)

**Comment by ndegendogo** [comment]
@di-sc nice idea your formula.

> The allowed Activity can be higher if the infection rate is low

Of course this reasoning is true. However, measuring the contact rate through cwa could give users a false sense of safety, as long as the **adoption rate** of cwa is not so high.
Currently the ratio of people sharing their keys to new infections is only about 1:5 or 1:6. So at the moment you can safely assume that several (or even most?) of your contacts will not be detectable with this method.

**Comment by di-sc** [comment]
@ndegendogo You mean some users could misuse the tool to check how far they can go with their activity until it shows the yellow notifiction? Good point, absolutely!

Some thoughts:
- The adoption rate could compensated by including it into the calculation or the threshold.
- The feature (as all features that make the app more responsive to the user) may make it more attractive for the user to install the app and increase the adoption rate.
- If the CWA green status would be made required to attend to events it would be not so attractive to reach the limit of a yellow status. As a side effect this would also increase the adoption rate, I think.

**Comment by ndegendogo** [comment]
@di-sc yes, you are right.
No matter what feature, there will always be silly people who misunderstand the purpose and misuse it. *Hopefully only very few such silly people*. And I think the majority indeed behaves prudent and will understand a high contact counter as a warning to re-think their behaviour and change it if possible.

**Comment by di-sc** [comment]
Until the ENF will support providing such contact statistic data it might be considered to use the [Event Notification](https://github.com/corona-warn-app/cwa-documentation/blob/master/event_registration.md) feature as a start. As I understand [Event Notification](https://github.com/corona-warn-app/cwa-documentation/blob/master/event_registration.md) will be released shortly and does not have these ENF restrictions!?

**Comment by Ein-Tim** [comment]
@di-sc 

Could you clarify what you would like to show from the event registration? 
Since it's part of the Corona-Warn-App, this feature is also restricted to the https://developer.apple.com/contact/request/download/Exposure_Notification_Addendum.pdf:

If it would break the rules, the update would risk to get blocked form the Stores, see https://www.bbc.com/news/technology-56713017

**Comment by di-sc** [comment]
@Ein-Tim It was only an idea. I never thought that having anonymous statistic data about my contacts would be a problem. But okay then that's not the way to go.

**Comment by Ein-Tim** [comment]
@di-sc I think nothing speaks about anonymous statistic data about your contacts, but not sure to what extend the even registration feature could help here.
